Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
Am I correct saying that `--shell` if your shell is Bash/ZSH or `--shell=bash` will preserve the same behaviour as there is currently? If so, I think that's enough to just require `--shell=bash` rather than a `--shell=none` which implies the same.